hikari/files.py

```diff
@@ -127,7 +127,7 @@
 - `aiohttp.StreamReader`
 """
 
-Resourceish = typing.Union["Resource[AsyncReader]", Pathish, Rawish]
+Resourceish = typing.Union["Resource", Pathish, Rawish]
```

Comment:
what are the implications of letting the Resource be treated as `Resource[typing.Any]` here as opposed to `Resource[AsyncReader]`

Reply:
I changed this because `URL` is a `Resource[WebReader]` and passing it as an attachment made mypy complain. I just removed the `AsyncReader` part.

Further more, `Resource` is defined like this:
```py
class Resource(typing.Generic[ReaderImplT], abc.ABC):
    ...
```
and `ReaderImplT` being: 
```py
ReaderImplT = typing.TypeVar("ReaderImplT", bound="AsyncReader")
```
which is already bound to `AsyncReader`, so I think it should be fine. Then again, if you think it should be changed to something else, then I will change it :smile: